### PR TITLE
Cleanups up and fixes docs in View class.

### PR DESCRIPTION
**Pull request by pavkam**
This PR does not introduce functional changes. It merely modernizes one class. If this PR is accepted more code can be modernized subsequently.

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [x] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by tig — COMMENTED** [review]
My OCD and desire to always be removing code loves all this.

My fear of regressions does not. 

I think we need more eyes on this before we merge it. I reviewed it fairly carefully, but I'd like @tzind and @bdisp to do the same.

**Comment by tznind** [comment]
Was this done with a tool? e.g. ReSharper? That would help my confidence in there being no mistakes a lot.

**Comment by BDisp** [comment]
> Was this done with a tool? e.g. ReSharper? That would help my confidence in there being no mistakes a lot.

I think so.

File: Terminal.Gui/Core/View.cs
Comment (tznind) [inline Terminal.Gui/Core/View.cs:2026]: I think this reduces readability.  There are compiler rules for `var` use that divide into 3 classifications:

- [csharp_style_var_for_built_in_types](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008#csharp_style_var_for_built_in_types) 
- [csharp_style_var_when_type_is_apparent](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008#csharp_style_var_when_type_is_apparent)
- [csharp_style_var_elsewhere](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008#csharp_style_var_elsewhere)

I think the first 2 are great but the third covers this instance.  It is when you can't tell by looking at the line what Type of variable you have allocated (without mousing over).  Old line you can tell that `tabIndexes` is views but new line is less clear.

 `csharp_style_var_elsewhere`

**Comment by tznind** [comment]
Ok I've gone through it and all the changes look good and correct.  Takes quite a while to comb through though, would be a big task to validate it across whole codebase.

**Comment by pavkam** [comment]
> Was this done with a tool? e.g. ReSharper? That would help my confidence in there being no mistakes a lot.

I use Rider on a Mac. The IDE does highlight the issues but I do manually validate whether a change is OK or not. I have not acted upon all its hints as that would be too intrusive.

File: Terminal.Gui/Core/View.cs
Comment (BDisp) [reply Terminal.Gui/Core/View.cs:2026]: `Roslynator` normally advice to use `Type` instead of `var` because is more faster and avoid confusing. I also use `var` a lot but change from `Type` to `var` wouldn't be necessary. But on overall I like the changes but I need review it careful.

**Comment by BDisp** [comment]
> I use Rider on a Mac. The IDE does highlight the issues but I do manually validate whether a change is OK or not. I have not acted upon all its hints as that would be too intrusive.

That a good point. Really Rider show many unnecessary suggestions. I'm glad you didn't accept all them. Good work.

File: Terminal.Gui/Core/View.cs
Comment (pavkam) [reply Terminal.Gui/Core/View.cs:2024]: I've switched that back to an explicit type and adjusted the Rider rules to account for that rule (though it still insists it's "apparent").

**Comment by pavkam** [comment]
@BDisp @tig @tznind -- if you think it makes sense to go around cleaning up more of the source code, I am glad to do so. Gives me an opportunity to figure out the code.

I was looking into creating a TUI toolkit before for my own needs but it makes more sense to modernize parts of this project and continue from there.

**Comment by BDisp** [comment]
> @BDisp @tig @tznind -- if you think it makes sense to go around cleaning up more of the source code, I am glad to do so. Gives me an opportunity to figure out the code.
> 
> I was looking into creating a TUI toolkit before for my own needs but it makes more sense to modernize parts of this project and continue from there.

For me I'm glad for your interesting on improvement the `Terminal.Gui`. You only must understand that you have to wait until all the changes are careful checked before @tig merge it. You are welcome.

**Comment by tig** [comment]
For the record, I'm still a tiny bit nervous about accepting this PR, but I really appreciate @pavkam's efforts and welcome him as a contributor!

Thanks to both @BDisp  @tznind for reviewing. 

In the future, any 'cleanups' like this should double down on more unit tests to reduce the risk of regressions.

I'm going to re-review this asap, just to make myself even more comfy with the changes...

**Comment by tznind** [comment]
> I was looking into creating a TUI toolkit before for my own needs but it makes more sense to modernize parts of this project and continue from there.

Awesome :) welcome aboard

**Review by tig — APPROVED** [review]
Thanks again @pavkam! Welcome to the project.

**Pull request merged**
